pkg/mcp/sink/server_sink.go

```diff
@@ -73,12 +79,10 @@ func NewServer(srcOptions *Options, serverOptions *ServerOptions) *Server {
 func (s *Server) EstablishResourceStream(stream mcp.ResourceSink_EstablishResourceStreamServer) error {
 	// TODO support receiving configuration from multiple sources?
 	// TODO MVP - limit to one connection at a time?
-	if !atomic.CompareAndSwapInt64(&s.connections, 0, 1) {
-		return errors.New("TODO limited to one connection at a time")
-	}
-	defer atomic.AddInt64(&s.connections, -1)
 
-	// TODO - rate limit new connections?
+	if !s.newConnectionLimiter.Allow() {
```

Comment:
We probably don't want to completely reject a new connection if the rate is too high. Instead, I think slowing the incoming rate via `Wait()` may be more appropriate here as it would spread out the subsequent config pushes to clients as they establish connections.

Reply:
That seems very reasonable, when I looked at the API (locally on my GOPATH) I came a cross the `wait` function however it was [unexported](https://github.com/golang/time/blob/26559e0f760e39c24d730d3224364aef164ee23f/rate/rate.go#L227) since Gopkg.lock is pointing to an older time/rate api in golang.org/x. I will go ahead and update the vendor for golang time and change it to `Wait`.